src/main/java/seedu/address/logic/commands/FilterPetCommand.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Filters and lists all pets in address book whose attributes match the argument keywords.
+ * Keyword matching is case-insensitive.
+ */
+public class FilterPetCommand extends Command {
+    public static final String COMMAND_WORD = "filter-p";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters all pets who are tagged: "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " s/Cat";
+
+    private final Predicate<Pet> cPredicate;
```

Comment:
Name of these variables may be a little unclear in suggesting what the predicate is about?